react/features/subtitles/components/TranscriptionSubtitles.web.js

```diff
@@ -0,0 +1,93 @@
+// @flow
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * The type of the React {@code Component} props of
+ * {@link TranscriptionSubtitles}.
+ */
+type Props = {
+
+    /**
+     * Map of transcriptMessageID's with corresponding transcriptMessage.
+     */
+    transcriptMessages: Map<string, Object>
+};
+
+/**
+ * React {@code Component} which can display speech-to-text results from
+ * Jigasi as subtitles.
+ */
+class TranscriptionSubtitles extends Component<Props> {
+
+    /**
+     * Updates the transcription subtitles only if the Map of transcriptMessages
+     * change otherwise prevents the unnecessary re-render.
+     *
+     * @inheritdoc
+     * @param {Object} nextProps - The props passed to the component before
+     * rendering the component.
+     * @returns {boolean} - True if props of the component changes, else false.
+     */
+    shouldComponentUpdate(nextProps) {
+        return this.props.transcriptMessages !== nextProps.transcriptMessages;
```

Comment:
Now that the logic has come down to one map of messages being passed in, can you confirm if render gets called when you don't to? react internally should only be re-rendering on state/prop change.

Reply:
Yeah this will always return true now. Should be safe to remove `shouldComponentUpdate` now.